pkg/kubelet/cm/devicemanager/plugin/v1beta1/stub.go

```diff
@@ -159,6 +208,35 @@ func (m *Stub) Stop() error {
 	return m.cleanup()
 }
 
+func (m *Stub) Watch(kubeletEndpoint, resourceName, pluginSockDir string) {
+	for {
+		select {
+		// Detect a kubelet restart by watching for a newly created
+		// 'pluginapi.KubeletSocket' file. When this occurs, restart
+		// the device plugin server
+		case event := <-m.kubeletRestartWatcher.Events:
+			if event.Name == kubeletEndpoint && event.Op&fsnotify.Create == fsnotify.Create {
+				klog.InfoS("inotify: file created, restarting", "kubeletEndpoint", kubeletEndpoint)
+				if err := m.Restart(); err != nil {
+					klog.ErrorS(err, "Unable to restart server")
+					panic(err)
```

Comment:
should we retry maybe couple of times. I am thinking of scenario when kubelet just created the socket, but not serving yet. Or some other race like this. I had bad experience with file watchers on Windows where notificaiton come very early (not this socket, but in general) and not sure how they work on Linux.

Reply:
Sure, that seems like a reasonable concern. I will add a mechanism to retry. Thanks!